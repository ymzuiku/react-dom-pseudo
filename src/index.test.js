import 'jest-styled-components';
import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { toSnapshot } from 'react-test-renderer-helper';
import Adapter from 'enzyme-adapter-react-16';
import Index from '../src/index.lib';

Enzyme.configure({ adapter: new Adapter() });

it('Index styled-compoments', () => {
  const comp = renderer.create(<Index />);
  expect(toSnapshot(comp)).toMatchSnapshot();
});

it('Index initAndUnmount', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Index />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Index updateNoChange', () => {
  const wrapper = shallow(<Index />);
  const firstDebug = wrapper.debug();
  expect(firstDebug).toMatchSnapshot('firstDebug');
  wrapper.update();
  expect(wrapper.debug()).toEqual(firstDebug);
});

// it('Index onClick status snapshots', () => {
//   const wrapper = shallow(<Index />);

//   expect(wrapper.debug()).toMatchSnapshot('init');

//   wrapper.simulate('click');
//   wrapper.simulate('onMouseEnter');
//   expect(wrapper.debug()).toMatchSnapshot('add');
//   expect(wrapper.state()).toMatchSnapshot('add-state');
// });