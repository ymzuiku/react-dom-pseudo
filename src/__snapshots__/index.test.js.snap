// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index onClick status snapshots: add 1`] = `"<div onMouseEnter={[Function]} onMouseLeave={[Function]} onFocus={[Function]} onBlur={[Function]} style={{...}} onClick={[Function]} onMouseDown={[Function]} onMouseUp={[Function]} />"`;

exports[`Index onClick status snapshots: add-state 1`] = `
Object {
  "active": false,
  "focus": false,
  "hover": false,
  "visited": true,
}
`;

exports[`Index onClick status snapshots: init 1`] = `"<div onMouseEnter={[Function]} onMouseLeave={[Function]} onFocus={[Function]} onBlur={[Function]} style={{...}} onClick={[Function]} onMouseDown={[Function]} onMouseUp={[Function]} />"`;

exports[`Index styled-compoments 1`] = `
Object {
  "props": Object {
    "activeStyle": undefined,
    "alwayStyle": undefined,
    "disable": false,
    "disableStyle": undefined,
    "focusStyle": undefined,
    "hoverStyle": undefined,
    "linkStyle": undefined,
    "merge": true,
    "style": undefined,
    "visitedStyle": undefined,
  },
  "state": Object {
    "active": false,
    "focus": false,
    "hover": false,
    "visited": false,
  },
  "status": undefined,
  "zip": <div
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={Object {}}
  />,
}
`;

exports[`Index updateNoChange: firstDebug 1`] = `"<div onMouseEnter={[Function]} onMouseLeave={[Function]} onFocus={[Function]} onBlur={[Function]} style={{...}} onClick={[Function]} onMouseDown={[Function]} onMouseUp={[Function]} />"`;
